unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Common/AssetPromise.cs

```diff
@@ -145,12 +155,17 @@ protected void OnLoadFailure(Exception exception)
             if (DataStore.i.common.isApplicationQuitting.Get())
                 return;
 #endif
-            
             CallAndClearEvents(false, exception);
             Cleanup();
         }
 
-        protected virtual bool AddToLibrary() { return library.Add(asset); }
+        protected virtual IEnumerator AddToLibrary(Action<bool> OnComplete)
```

Comment:
IMO AddToLibrary should be an "atomic" method and not async/coroutine. It shouldnt also be capable of "failing" and the need of the callback should be removed.

We should be able to do the texture compression throttling when we retrieve the texture and leave all the flow as it is currently.